code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -575,6 +575,18 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 			src.splash_all_contents = 1
 		return
 
+	attackby(obj/item/I, mob/user)
```

Comment:
Would rather put this in the `afterattack` proc of sugar cubes than here. It's a pretty specific interaction so I don't think `drinks` items need to know about it, just let the sugar cube handle itself.

Also, putting it here means this won't work with beakers/water tanks/whatever. A slightly cleaner way of doing it would be to check `if (target.is_open_container() && target.reagents)` (where `target` is the thing you're clicking with the cube) in the cube's `afterattack`, which lets you dissolve the cubes in anything open that has a reagent holder.

Reply:
Totally valid criticism, in fact I wanted to do something similar with `attack`, but that only has a `mob/user`, and I wasn't sure if `afterattack` was sufficient. Also the reagents check is smarter, thanks!